### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by VelzckC0D3**
# ✨ Requested Implementations ✨

image
> - _Output_
image

# 💡 Thank you for your review! 💡

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,12 +1,19 @@
-require './person'
+require_relative 'person'
 
 class Student < Person
-  def initialize(age, classroom, _name = 'Unknown', parent_permission: true)
+  attr_accessor :classroom
```
Comment (cynthiainga) [inline student.rb]: - [x] Good job so far on the association but you just need to fix some issues and you are good to go, the `classroom` is a `setter` which means the `classroom` should be `attr_reader`. Kindly fix that 🙏🏾

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative 'rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (cynthiainga) [inline book.rb]: - [x] Getter and Setter for `@rentals` as for `@title` and `@author`, `rentals` should be `attr_accessor` as well. Kindly fix that 🙏🏾

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (cynthiainga) [inline rental.rb]: - [x] Getters and Setters for `@book` and `@person` as for `@date`, `book and person` should be `attr_accessor` as well. Kindly fix that 🙏🏾

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  def initialize(date, book, person)
+    @date = date
+
+    @book = book
+    @book.rentals << self
+
+    @person = person
```
Comment (cynthiainga) [inline rental.rb:10]: - [x] The rental should be adding itself to the `@person` the `belongs-to` association as you did for the book above, kindly fix that 😉

File: person.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
-require './nameable'
+require_relative 'nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (cynthiainga) [inline person.rb]: - [x] Getters and Setters for `@rentals`, the rentals should be `attr_accessor`, kindly fix that 🙏🏾

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @VelzckC0D3, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] The `has-many` / `belongs-to` relationships between `Classroom` and `Student` are created ✔️ 
- [ ] A person has many rentals ✔️ 
- [ ] A rental belongs to a book ✔️ 
- [ ] A book has many rentals ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can get the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (VelzckC0D3) [reply student.rb]: Thank you for noticing, It's done!

File: book.rb (hunk 2 above)
Comment (VelzckC0D3) [reply book.rb]: Done too ✨

File: rental.rb (hunk 3 above)
Comment (VelzckC0D3) [reply rental.rb]: ✔️ Checked

File: rental.rb (hunk 4 above)
Comment (VelzckC0D3) [reply rental.rb:10]: also, done, thank you for your review

File: person.rb (hunk 5 above)
Comment (VelzckC0D3) [reply person.rb]: corrected

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @VelzckC0D3 

Your project is complete! Great job for nailing it 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request ✔️ 
- Associations ✔️ 
- many/belongs to relationships✔️ 
- No linter errors✔️ 

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**